bin/hipcc

```diff
@@ -762,20 +775,24 @@ if ($HIP_PLATFORM eq "clang") {
     }
     # Do not pass -mllvm on Windows since there is a clang bug causing duplicate -mllvm options in clang -cc1 on Windows.
     # ToDo : remove restriction for Windows after clang bug is fixed.
-    if (!$funcSupp and $optArg ne "-O0" and not $isWindows) {
+    if (!$funcSupp and $optArg ne "-O0" and not $isWindows and $hasHIP) {
         $HIPCXXFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
         if ($needLDFLAGS and not $needCXXFLAGS) {
             $HIPLDFLAGS .= " -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false";
```

Comment:
Are these LDFLAGS needed based on this comment: https://github.com/ROCm-Developer-Tools/HIP/pull/2042#pullrequestreview-399934463 ?
Does hipcc on an object file as input with no source files work? (That was the attempted fix of #2042 )

Reply:
Yes. There is condition $hasHIP, which is true only if the command has .hip file or .cpp/.cc/.cxx files together with -x hip.

If there is only .o files, these options are not added.

Reply:
Ok. The flags are getting trickier for me to track. Maybe refactoring could help?
Should LDFLAGS be prefixed by `-Wl` to be passed to the linker if that's what they are meant for as the name might suggest?

Reply:
in LD flags, the options are intended for clang, not for lld. When -fgpu-rdc is passed, .o files are handled by clang to do device linking and codegen.
We need some minor fixes to check for -fgpu-rdc for LD flags. However the current priority is converge master and amd-master-branch, so we will fix this later.